src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -34,51 +24,210 @@
- * 
+ *
  * branch API test 5
+ *
+ * @see <a href="http://stackoverflow.com/questions/3641722/valid-characters-for-uri-schemes">Valid characters for URI schemes</a>
  */
 public class CodecUtils {
 
   public final static Logger log = LoggerFactory.getLogger(CodecUtils.class);
 
+  /**
+   * A description of an API type
+   */
   public static class ApiDescription {
-    String key;
-    String path; // {scheme}://{host}:{port}/api/messages
-    String exampleUri;
-    String description;
+    /**
+     * The string to use after {@link #PARAMETER_API}
+     * in URIs to select this API.
+     */
+    public final String key;
 
+    /**
+     * The path to reach this API
+     */
+    public final String path; // {scheme}://{host}:{port}/api/messages
+
+    /**
+     * An example URI to reach this API
+     */
+    public final String exampleUri;
+
+    /**
+     * The description of this API
+     */
+    public final String description;
+
+    /**
+     * Construct a new API description.
+     *
+     * @param key {@link #key}
+     * @param uriDescription {@link #path}
+     * @param exampleUri {@link #exampleUri}
+     * @param description {@link #description}
+     */
     public ApiDescription(String key, String uriDescription, String exampleUri, String description) {
       this.key = key;
       this.path = uriDescription;
       this.exampleUri = exampleUri;
       this.description = description;
     }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      ApiDescription that = (ApiDescription) o;
+      return Objects.equals(key, that.key) && Objects.equals(path, that.path) && Objects.equals(exampleUri, that.exampleUri) && Objects.equals(description, that.description);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(key, path, exampleUri, description);
+    }
   }
 
+  /**
+   * The string to be used to specify the API in URIs,
+   * with leading and trailing slash.
+   */
   public final static String PARAMETER_API = "/api/";
+
+  /**
+   * The string to be used in URIs to specify the API
+   */
   public final static String PREFIX_API = "api";
 
   // mime-types
+  /**
+   * The MIME type used to specify JSON data.
+   *
+   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types">MIME types</a>
+   */
   public final static String MIME_TYPE_JSON = "application/json";
 
-  private transient static Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd HH:mm:ss.SSS").disableHtmlEscaping().create();
-  private transient static Gson prettyGson = new GsonBuilder().setDateFormat("yyyy-MM-dd HH:mm:ss.SSS").setPrettyPrinting().disableHtmlEscaping().create();
+  /**
+   * Whether we are using the GSON JSON backend or not. If false,
+   * use Jackson.
+   * TODO Replace with enum to allow extension for multiple backends

```

Comment:
when will we be using gson and when jackson ?

Reply:
Right now Gson is the selected backend for compatibility reasons, I'd like to switch to Jackson though because it's more flexible. This boolean switch was made in case there's an issue we discover with the Jackson backend down the line so we can quickly switch over to Gson with a one-line hotfix